consensus/stake.go

```diff
@@ -38,18 +37,19 @@ func newWeightCounter(vv Validators) *WeightCounter {
 	return &WeightCounter{
 		validators:   vv,
 		quorum:       vv.Quorum(),
+		antiQuorum:   vv.TotalWeight() - vv.Quorum(),
 		alreadyVoted: make([]bool, vv.Len()),
 		sum:          0,
+		antiSum:      0,
 	}
 }
 
-// CountVoteByID validator and return true if it hadn't counted before.
 func (s *WeightCounter) CountVoteByID(v ValidatorID) bool {
 	validatorIdx := s.validators.GetIdx(v)
 	return s.CountVoteByIndex(validatorIdx)
 }
 
-// CountVoteByIndex validator and return true if it hadn't counted before.
+// CountVoteByIndex increases the vote sum by validator's weight if the validator hasn't voted before, returning the status of the vote
```

Comment:
nit: I would perhaps favor long forms of negative predicates (has not instead of hasn't). Leave it be if such forms are prominent in the whole codebase.